File: ts/tests/hooks/testutil.ts
import { readFileSync } from "node:fs";
import path from "node:path";
import { specRoot } from "../../src/index";

/** Reads and parses a golden fixture for `host`, by file name (e.g. "PreToolUse.block.json"). */
export function fixture(host: string, name: string): unknown {
  const raw = readFileSync(path.join(specRoot(), "fixtures", "hosts", host, name), "utf8");
  return JSON.parse(raw);
}

File: py/tests/hooks/testutil.py
"""Shared fixture-loading helper for the hooks test suite."""

from __future__ import annotations

import json
import os

from axon import spec_root


def fixture(host: str, name: str) -> object:
    """Reads and parses a golden fixture for `host`, by file name (e.g. "PreToolUse.block.json")."""
    path = os.path.join(spec_root(), "fixtures", "hosts", host, name)
    with open(path, encoding="utf-8") as f:
        return json.load(f)
